python/lsst/display/firefly/firefly.py

```diff
@@ -208,67 +249,110 @@ def _getEvent(self):
     #
     # Set gray scale
     #
+
     def _scale(self, algorithm, min, max, unit=None, *args, **kwargs):
-        stretchAlgorithms = ("Linear", "Log", "LogLog", "Equal", "Squared", "Sqrt")
+        stretch_algorithms = ('linear', 'log', 'loglog', 'equal', 'squared', 'sqrt',
+                              'asinh', 'powerlaw_gamma')
+        interval_methods = ('percent', 'maxmin', 'absolute', 'zscale', 'sigma')
+        #
         #
         # Normalise algorithm's case
         #
         if algorithm:
-            algorithm = dict((a.lower(), a) for a in stretchAlgorithms).get(algorithm.lower(), algorithm)
+            algorithm = dict((a.lower(), a) for a in stretch_algorithms).get(algorithm.lower(), algorithm)
 
-            if not algorithm in stretchAlgorithms:
+            if algorithm not in stretch_algorithms:
                 raise FireflyError('Algorithm %s is invalid; please choose one of "%s"' %
-                                   (algorithm, '", "'.join(stretchAlgorithms)))
+                                   (algorithm, '", "'.join(stretch_algorithms)))
             self._stretchAlgorithm = algorithm
+        else:
+            algorithm = 'linear'
 
-        if min == "minmax":
-            unit = "Percent"
+        if min == 'minmax':
+            interval_type = 'percent'
+            unit = 'percent'
             min, max = 0, 100
-        elif min == "zscale":
-            pass
+        elif min == 'zscale':
+            interval_type = 'zscale'
 
```

Comment:
You need an else clause to set `interval_type = None` (otherwise `display.scale("log", 0, 100)` generates an exception)